rpcserver.go

```diff
@@ -1168,57 +1175,87 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
 func (r *rpcServer) DecodeProof(ctx context.Context,
 	in *taprpc.DecodeProofRequest) (*taprpc.DecodeProofResponse, error) {
 
-	if len(in.RawProof) == 0 {
+	if len(in.RawProof) < 4 {
```

Comment:
I think `proof.PrefixMagicBytesLength` would be clearer here, rather than `4`. But also see my other comment on the idea of not exporting the magic bytes concept out of the `proof` package.

Reply:
Makes sense, changed.